README.md

```diff
@@ -190,19 +190,230 @@ path : /path/to/executable-file
 --
 ```
 
-### `assert_files_equal`
+### `assert_file_owner`
 
-Fail if the given files are not the same.
+Fail if given user is not the owner of the given file.
 
 ```bash
-@test 'assert_files_equal()' {
-  assert_files_equal /path/to/file1 /path/to/file2
+@test 'assert_file_owner() {
+  assert_file_owner $owner /path/to/owner
 }
 ```
 
 On failure, the path is displayed.
 
 ```
+-- user is not the owner of the file --
+path : /path/to/notowner
+owner : user
+--
+```
+
+### `assert_not_file_owner`
+
+Fail if given user is the owner of the given file.
+
+```bash
+@test 'assert_not_file_owner() {
+  assert_not_file_owner $owner /path/to/notowner
+}
+```
+
+On failure, the path is displayed.
+
+```
+-- Fail if given user is the owner, but it was expected not to be --
+path : /path/to/owner
+owner : $owner
+--
+```
+
+### `assert_file_permission`
+
+Fail if given file does not have permissions 777.
+
+```bash
+@test 'assert_file_permission() {
+  assert_file_permission $permission /path/to/permission
+}
+```
+
+On failure, the path is displayed.
+
+```
+-- given file does not have permissions 777 --
+path : /path/to/nopermission
+permission: $permission
+--
+```
+
+### `assert_no_file_permission`
+
+Fail if given file has permissions 777.
+
+```bash
+@test 'assert_no_file_permission() {
+  assert_no_file_permission $permission /path/to/nopermission
+}
+```
+
+On failure, the path is displayed.
+
+```
+-- given file has permissions 777, but it was expected not to have --
+path : /path/to/permission
+permission : $permission
+--
+```
+
+### `assert_zero`
+
+Fail if file is not zero byte.
+
+```bash
+@test 'assert_zero() {
+  assert_zero /path/to/zerobyte
+}
+```
+
+On failure, the path is displayed.
+
+```
+-- file is not zero byte --
+path : /path/to/notzerobyte
+--
+```
+
+### `assert_not_zero`
+
+Fail if file size is zero byte.
+
+```bash
+@test 'assert_not_zero() {
+  assert_not_zero /path/to/notzerobyte
+}
+```
+
+On failure, the path is displayed.
+
+```
+-- file is 0 byte, but it was expected not to be --
+path : /path/to/zerobyte
+--
+```
+
+### `assert_file_set_group_id`
+
+Fail if group id is not set.
+
+```bash
+@test 'assert_file_set_group_id() {
+  assert_file_set_group_id /path/to/groupidset
+}
+```
+
+On failure, the path is displayed.
+
+```
+-- group id is not set --
+path : /path/to/groupidnotset
+--
+```
+
+### `assert_file_not_set_group_id`
+
+Fail if group id is set.
+
+```bash
+@test 'assert_file_not_set_group_id() {
+  assert_file_not_set_group_id /path/to/groupidnotset
+}
+```
+
+On failure, the path is displayed.
+
+```
+-- group id is set, but it was expected not to be --
+path : /path/to/groupdidset
+--
+```
+
+### `assert_file_set_user_id`
```

Comment:
please rename to `assert_file_user_id_set` and `assert_file_not_user_id_set` everywhere.